pypeit/spectrographs/keck_deimos.py

```diff
@@ -703,24 +708,64 @@ def _grating_orientation(slider, ruling, tilt):
-                                      900: [ 0.141,  0.000, 5.6e-4, -0.134],
-                                     1200: [ 0.145,  0.055, 5.6e-4, -0.181],
-                                  'other': [ 0.145,  0.000, 5.6e-4, -0.182] },
-                              4: {    600: [-0.065,  0.063, 6.9e-4, -0.298],
-                                      831: [-0.034,  0.060, 6.9e-4, -0.196],
-                                      900: [-0.064,  0.083, 6.9e-4, -0.277],
-                                     1200: [-0.052,  0.122, 6.9e-4, -0.294],
-                                  'other': [-0.050,  0.080, 6.9e-4, -0.250] } }
+        orientation_coeffs = {3: {    600: [ 0.145, -0.008, 5.6e-4, -0.146],
+                                      831: [ 0.143,  0.000, 5.6e-4, -0.018],
+                                      900: [ 0.141,  0.000, 5.6e-4, -0.118],
+                                     1200: [ 0.145,  0.055, 5.6e-4, -0.141],
+                                  'other': [ 0.145,  0.000, 5.6e-4, -0.141] },
+                              4: {    600: [-0.065,  0.063, 6.9e-4, -0.108],
+                                      831: [-0.034,  0.060, 6.9e-4, -0.038],
+                                      900: [-0.064,  0.083, 6.9e-4, -0.060],
+                                     1200: [-0.052,  0.122, 6.9e-4, -0.110],
+                                  'other': [-0.050,  0.080, 6.9e-4, -0.110] } }
+
+        # Orientation coefficients meant for observations taken Pre-2016 Servicing
+        # orientation_coeffs = {3: {    600: [ 0.145, -0.008, 5.6e-4, -0.182],
+        #                               831: [ 0.143,  0.000, 5.6e-4, -0.182],
+        #                               900: [ 0.141,  0.000, 5.6e-4, -0.134],
+        #                              1200: [ 0.145,  0.055, 5.6e-4, -0.181],
+        #                           'other': [ 0.145,  0.000, 5.6e-4, -0.182] },
+        #                       4: {    600: [-0.065,  0.063, 6.9e-4, -0.298],
+        #                               831: [-0.034,  0.060, 6.9e-4, -0.196],
+        #                               900: [-0.064,  0.083, 6.9e-4, -0.277],
+        #                              1200: [-0.052,  0.122, 6.9e-4, -0.294],
+        #                           'other': [-0.050,  0.080, 6.9e-4, -0.250] } }
 
         # Return calbirated roll, yaw, and tilt
         return orientation_coeffs[slider][_ruling][0], \
                 orientation_coeffs[slider][_ruling][1], \
                 tilt*(1-orientation_coeffs[slider][_ruling][2]) \
                     + orientation_coeffs[slider][_ruling][3]
 
+
+
+    def get_amapbmap(self, filename):
+        """
+        Select the pre-grating (amap) and post-grating (bmap) maps according to the slider.
+        """
+        hdu = fits.open(filename)
+
+        # Grating slider
+        slider = hdu[0].header['GRATEPOS']
+
+        mp_dir = resource_filename('pypeit', 'data/static_calibs/keck_deimos/')
+
+        if slider in [3,4]:
```

Comment:
IDL save files are in general not a good long term solution for storing data. I realize that there exists a method to read them in, but we should be reading in these maps from the save files, and then writing them out to fits or another stable binary format, and then reading those in in PypeIt. My suggestion is that you write a utility routine in core that does this and then push the converted fits files to the pypeit data repository.

Reply:
OK. Done! I wrote a small method in `core/convert_DEIMOSsavfiles.py`. Please let me know if would be better to put it in another folder. This small script converted the .sav file in .fits files, which are now in `data/static_calibs/keck_deimos/`